marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/survivorship/merging/base.xqy

```diff
@@ -2357,45 +2354,52 @@ declare variable $locked-uris-map as map:map := map:map();
 
 declare function merge-impl:locked-uris() {
   if (map:contains($locked-uris-map, "runAlready")) then
-    map:get($locked-uris-map, "lockedURIs")
+    $locked-uris-map
   else
-    let $locked-uris := (
-      map:get($locked-uris-map, "lockedURIs"),
-      for $host-status in xdmp:host-status(xdmp:hosts())
-      let $host-id := $host-status/host:host-id
-      for $transaction-id in $host-status/host:transactions/host:transaction/host:transaction-id
-      return
-        (: Surround in try/catch since transaction could have closed since status or call to other host could timeout :)
-        try {
-          (: We only care about locks by mastering. We'll still need to wait on outside transactions. :)
-          xdmp:transaction-locks($host-id, $transaction-id)/host:write[fn:starts-with(., $lock-task-prefix)]/fn:substring-after(., $lock-task-prefix)
-        } catch ($e) {()}
-    )
+    let $transaction-id := xdmp:transaction()
+    let $locked-uris := fn:distinct-values(
+                        for $host-id in xdmp:hosts()
+                        let $check-transactions := xdmp:host-status($host-id)/host:transactions/host:transaction[host:transaction-mode = "update"]/host:transaction-id[. ne $transaction-id]
+                        for $check-transaction in $check-transactions
+                        return xdmp:spawn-function(function() {merge-impl:locked-uris($host-id, $check-transaction)}, map:map() => map:with("update","false") => map:with("priority","higher") => map:with("result",fn:true()))
```

Comment:
I think a comment explaining why spawn-function is used here would help as well.

Reply:
Changed to invoke-function and provided a comment in latest push.